bug: float division by zero throws error instead of returning math.INF

## Description
Dividing a float by zero throws a runtime error `E5001: division by zero` instead of returning `math.INF` as expected in IEEE 754 compliant languages.

## Expected Behavior
Since EZ has `math.INF` and `math.NEG_INF` constants and a `math.is_inf()` function, float division by zero should return `math.INF` (or `math.NEG_INF` for negative dividends) rather than throwing an error.

## Reproduction
```ez
import @std
using std

do main() {
    temp f_divisor float = 0.0
    temp f_result float = 10.0 / f_divisor  // Throws E5001 runtime error
    println("Result: ${f_result}")
}
```

## Actual Behavior
```
error[E5001]: division by zero
  --> test.ez:7:32
   |
7  |     temp f_result float = 10.0 / f_divisor
   |                                ^ error occurred here
```

## Why This Is A Bug
- The language explicitly supports infinity values via `math.INF`, `math.NEG_INF`
- The language has `math.is_inf()` to check for infinity
- IEEE 754 defines `x / 0.0 = INF` for positive x, `-INF` for negative x
- Users cannot create infinity values at runtime through normal arithmetic operations

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -1664,9 +1664,7 @@ func evalFloatInfixExpression(operator string, left, right Object, line, col int
 	case "*":
 		return &Float{Value: leftVal * rightVal}
 	case "/":
-		if rightVal == 0 {
-			return newErrorWithLocation("E5001", line, col, "division by zero")
-		}
+		// Float division by zero returns +Inf, -Inf, or NaN per IEEE 754
 		return &Float{Value: leftVal / rightVal}
 	case "<":
 		return nativeBoolToBooleanObject(leftVal < rightVal)

